code/controllers/subsystems/initialization/materials.dm

```diff
@@ -76,11 +76,11 @@ SUBSYSTEM_DEF(materials)
 		return
 	materials =         list()
 	materials_by_name = list()
-	for(var/mtype in subtypesof(/decl/material))
-		var/decl/material/new_mineral = mtype
-		if(!initial(new_mineral.name))
+	var/list/material_decls = decls_repository.get_decls_of_subtype(/decl/material)
+	for(var/mtype in material_decls)
+		var/decl/material/new_mineral = material_decls[mtype]
+		if(!new_mineral.name)
```

Comment:
Are there any non-abstract materials with no name? I thought that violated a required unit test?

Reply:
Looks like it doesn't, but it definitely should, since everything else is autogenerated based on `name`.